src/components/NewTimerModal.tsx

```diff
@@ -1,104 +1,105 @@
 import { useDisclosure } from '@chakra-ui/react'
 import {
-  Modal,
-  ModalOverlay,
-  ModalContent,
-  ModalHeader,
-  ModalFooter,
-  ModalBody,
-  ModalCloseButton,
+
   Button
 } from '@chakra-ui/react'
 import { AddIcon } from '@chakra-ui/icons'
-import { useState, useEffect } from 'react'
-import { Timer, NewTimerModalProps } from '../types/app-types'
+import { useEffect, useState, } from 'react'
+import { NewTimerModalProps, IncomingTimerProps, TimerConfig } from '../types/app-types'
 import NamingPhase from './ModalSteps/NamingPhase'
 import GamingPhase from './ModalSteps/GamingPhase'
 import TimingPhase from './ModalSteps/TimingPhase'
+import { GameKey } from './Timer/timerStyles'
+import { Modal } from './Modal'
 
 enum Phase {
   SetName,
   SetGameType,
-  SetCountdown
+  SetCountdown,
+  Complete
+}
+
+const initialParams = {
+  name: '',
+  game: '' as GameKey,
+  initialTime: { minutes: 0, seconds: 0 },
 }
 
 const NewTimerModal: React.FC<NewTimerModalProps> = ({ onModalComplete }) => {
   const { isOpen, onOpen, onClose } = useDisclosure()
   const [phase, setPhase] = useState<Phase>(Phase.SetName)
-  const [newTimerParams, setNewTimerParams] = useState<Timer>({
-    title: '',
-    game: '',
-    initialTime: '',
-    timeAtPause: ''
-  })
-
-  const initialTimerParams = {
-    title: '',
-    game: '',
-    initialTime: '',
-    timeAtPause: ''
+  const [newTimerParams, setNewTimerParams] = useState<TimerConfig>(initialParams)
+
+  const resetModal = () => {
+    onClose()
+    setPhase(Phase.SetName)
+    setNewTimerParams(initialParams)
+  }
+
+  const onBack = () => {
+    if (phase === Phase.SetName) {
+      return
+    }
+    setPhase((prevPhase) => prevPhase - 1)
+  }
+
+  const nextPhase = () => {
+    const transitions: Partial<Record<Phase, Phase>> = {
+      [Phase.SetName]: Phase.SetGameType,
+      [Phase.SetGameType]: Phase.SetCountdown,
+      [Phase.SetCountdown]: Phase.Complete
+    }
```

Comment:
This is a nice pattern for transitioning between steps. Can use the same config for going backwards, but I didn't implement that